test_board.py

```diff
@@ -74,8 +74,78 @@ def test_build_board(self):
 
         with self.subTest('should build a board with the width matching the input number'):
             self.test_board = Board(5)
-            expected_board = [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10],
-                              [11, 12, 13, 14, 15], [16, 17, 18, 19, 20], [21, 22, 23, 24, 25]]
+            expected_board = [[1, 2, 3, 4, 5],
+                              [6, 7, 8, 9, 10],
+                              [11, 12, 13, 14, 15],
+                              [16, 17, 18, 19, 20],
+                              [21, 22, 23, 24, 25]]
 
             result = self.test_board.get_board()
             self.assertEqual(result, expected_board)
+
+    def test_when_updating_board(self):
+        with self.subTest("should place player's mark on desired cell"):
+            self.test_board = Board(5)
+            self.test_board.update_board("x", 1)
+            expected_output = [["x", 2, 3, 4, 5],
+                               [6, 7, 8, 9, 10],
+                               [11, 12, 13, 14, 15],
+                               [16, 17, 18, 19, 20],
+                               [21, 22, 23, 24, 25]]
+            self.assertEqual(self.test_board.get_board(), expected_output)
+
+    def test_get_board_state(self):
+        self.test_board = Board(5)
+        with self.subTest('game is not finished if there are no winners'):
+            self.test_board._board = [[1, 2, 3, 4, 5],
+                                      [6, 7, 8, 9, 10],
+                                      [11, 12, 13, 14, 15],
+                                      [16, 17, 18, 19, 20],
+                                      [21, 22, 23, 24, 25]]
+            result = self.test_board.get_board_winner_status()
+            self.assertEqual(result, WinnerStatus.ONGOING)
+
+        with self.subTest('should check the state of the rows'):
+            self.test_board._board = [['x', 'x', 'x', 'x', 'x'],
+                                      [6, 7, 8, 9, 10],
+                                      [11, 12, 13, 14, 15],
+                                      [16, 17, 18, 19, 20],
+                                      [21, 22, 23, 24, 25]]
+            result = self.test_board.get_board_winner_status()
+            self.assertEqual(result, WinnerStatus.WINNER_X)
+
+        with self.subTest('should check the state of the columns'):
+            self.test_board._board = [['x', 2, 3, 4, 5],
```

Comment:
Same for this one, should it be "should return X as the winner when X updates all cells in a column"?